### PR TITLE
API fix : EthDoCall and DoCall for compatibility

### PR DESCRIPTION
**Pull request by kjeom**
## Proposed changes

- Issue
  - If there is neither `GasPrice` nor `GasFeeCap` in API arguments of DoCall, EthDoCall, an error occured.
  - It causes incompatibility with legacy services.
- Fix
  - Deleted the condition.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- #1511

### CODE REVIEW

File: api/api_ethereum.go [hunk 1]
```diff
@@ -1225,8 +1225,6 @@ func (args *EthTransactionArgs) ToMessage(globalGasCap uint64, baseFee *big.Int,
 			gasPrice = args.GasPrice.ToInt()
 		} else if args.MaxFeePerGas != nil {
 			gasPrice = args.MaxFeePerGas.ToInt()
-		} else {
```
Comment (jimni1222) [inline api/api_ethereum.go:1228]: Can you fix L1223 comment? I think this if condition includes EthTxTypeCompatible hard fork (support EIP-1559) also.

Comment (kjeom) [reply api/api_ethereum.go:1228]: Thanks!

**Review by jimni1222 — DISMISSED** [review]

**Review by bobpkr — DISMISSED** [review]

File: blockchain/state_transition.go [hunk 2]
```diff
@@ -135,7 +135,7 @@ type kerror struct {
 
 // NewStateTransition initialises and returns a new state transition object.
 func NewStateTransition(evm *vm.EVM, msg Message) *StateTransition {
-	// before kip71 hardfork, effectiveGasPrice is unitPrice
```
Comment (hqjang-pepper) [inline blockchain/state_transition.go:138]: Don't we need to change kip71 hardfork to magma hardfork here too?

Comment (hqjang-pepper) [reply blockchain/state_transition.go:138]: Oh this change is included in https://github.com/klaytn/klaytn/pull/1514. Resolved.

**Review by hqjang-pepper — DISMISSED** [review]

**Review by aidan-kwon — COMMENTED** [review]
In v1.8.4. DoCall set `gasPrice` to `defaultGasPrice (25 ston)` if a user doesn't input it. But, the current code set it to `0` as a default. How about using `lowwerBound` when  `baseFee.Cmp(new(big.Int).SetUint64(params.ZeroBaseFee)) == 0` and `baseFee` when `baseFee.Cmp(new(big.Int).SetUint64(params.ZeroBaseFee)) != 0`?

**Comment by kjeom** [comment]
Right... I think it is better to discuss in lower/upper gasPrice API PR.
The lowerBoundBaseFee can be loaded from governance in the PR.

**Comment by aidan-kwon** [comment]
> Right... I think it is better to discuss in lower/upper gasPrice API PR. The lowerBoundBaseFee can be loaded from governance in the PR.

Okay. The semantic is different from the previous version, but `0` seems good, also. Let's discuss it later.

**Review by aidan-kwon — APPROVED** [review]

**Review by jimni1222 — APPROVED** [review]

**Review by hqjang-pepper — APPROVED** [review]

**Review by bobpkr — APPROVED** [review]

**Review by aidan-kwon — APPROVED** [review]

**Pull request merged**